src/Entity/FieldableMonetizationEntityBaseTrait.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_m10n\Entity;
+
+use Drupal\apigee_edge\Entity\FieldableEdgeEntityBaseTrait;
+
+/**
+ * Trait that allows to make Apigee Monetization entities fieldable.
+ *
+ * Contains implementations that were only available for content entities.
+ *
+ * A fieldable Edge entity's properties that are being exposed as base fields
+ * should not be modified by using the original property setters (inherited
+ * from the wrapper SDK entity), those should be modified through the field
+ * API because that can keep field and property values in sync.
+ *
+ * @see \Drupal\Core\Entity\ContentEntityBase
+ * @see \Drupal\Core\Entity\FieldableEntityStorageInterface
+ * @see \Drupal\apigee_edge\Entity\FieldableEdgeEntityInterface
+ */
+trait FieldableMonetizationEntityBaseTrait {
+
+  use FieldableEdgeEntityBaseTrait {
+    set as edgeSet;
+  }
+
+  /**
+   * This is mostly copied from `FieldableEdgeEntityBaseTrait` but we are adding
```

Comment:
Could you add this feature without code duplication? :skull_and_crossbones:
For example by introducing a base class for mint entities that extends the inherited set() method from this trait?

Reply:
There have been some recent changes to `apigee_edge` that breaks what we are doing here. We will see what we can to about this when we pull in those changes and make the necessary fixes.